### DISCUSSION TITLE
import not working working properly in webpack.

### DISCUSSION THREAD

**Discussion by spikeHollow** [vercel/next.js#45778] [Help]
I'm using next-js 13 with app directory enabled.
I've this function
```javascript
export const ComponentImporter = (compString: Component, props: any = {}) => {
  return import("@/components/" + compString)
    .then(comp => {
      const rcomp = React.createElement(comp, props);
      return React.createElement(ChakraProvider, {theme}, rcomp);
    });
};
```
in `src\components\index.ts`.

When I import it, like these:
```javascript
import { ComponentImporter } from "../components/index";
```
Instead of giving the `ComponentImporter` function, it gives this object. 
`{$$typeof: Symbol(react.module.reference), filepath: '...', name: 'ComponentImporter', async: false}`.

Why is it giving this?

How to import the function?

**Comment by ciruz** [comment]
Hey,

when you import a component the output would be a Promise.

So try it with an ``async`` loader and use ``await import``.

See this link and also checkout the example there (``fuse.js``):
- https://beta.nextjs.org/docs/optimizing/lazy-loading#example-loading-external-libraries
> External libraries can be loaded on demand using the [import()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) function. This example uses the external library fuse.js for fuzzy search. The module is only loaded on the client after the user types in the search input.

-  [import()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) 
> [Return value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#return_value)
Returns a promise [..]

**Reply by spikeHollow** [reply]
Actually, it doesn't work. I tried it using await import. It just wraps the object in Proxy. Here's the output.
```
Proxy {$$typeof: Symbol(react.module.reference), filepath: 'D:\Projects\next-projects\user-mod\src\components\index.ts', name: '*', async: true, then: {…}}
```

**Reply by ciruz** [reply]
Do you have a small simple repo with this error to reproduce?

**Reply by spikeHollow** [reply]
Sorry, I'm using completely different approach right now. I've already deleted that code.
Anyway, thanks for trying to help me.
